test/e2e/utils/file_utils.py

```diff
@@ -19,3 +20,39 @@ def create_random_file():
 
 def extract_file_name(file_path):
     return Path(file_path).name
+
+
+def verify_downloaded_file_contents(download, text):
+    file_path = download_path(download)
+    download.save_as(file_path)
+    try:
+        with open(file_path) as file:
+            assert file.read() == text
+    finally:
+        os.remove(file_path)
+
+
+def verify_downloaded_archive_contents(
+    archive_download, *args, match_all_args=True
+):
+    file_path = download_path(archive_download)
+    archive_download.save_as(file_path)
+    try:
+        with ZipFile(file_path) as zip_file:
+            file_names = zip_file.namelist()
+            if match_all_args:
+                assert len(file_names) == len(
+                    args
+                ), "Number of files in downloaded archive was incorrect"
+            for file_name in file_names:
+                with zip_file.open(file_name) as file:
+                    file_content = file.read().decode()
+                    assert (
+                        file_content in args
+                    ), f"Unexpected content of file in archive {file_content}"
+    finally:
+        os.remove(file_path)
+
+
+def download_path(download):
+    return f"{os.getcwd()}\\{download.suggested_filename}"
```

Comment:
Please use `os.path.join` here: `os.path.join(os.getcwd(), download_suggested_filename)`. It will take care of the path separator being the right one for the operating system.

Reply:
done